src/python/WMCore/MicroService/Unified/Common.py

```diff
@@ -80,12 +80,39 @@ def dbsInfo(datasets, dbsUrl):
     return datasetBlocks, datasetSizes, datasetTransfers
 
 
-def phedexValidBlocks(datasets, phedexUrl):
+def getPileupDatasetSizes(datasets, phedexUrl):
     """
     Given a list of datasets, find all their blocks with replicas
-    available, i.e., blocks that have valid files to be processed.
+    available, i.e., blocks that have valid files to be processed,
+    and calculate the total dataset size
     :param datasets: list of dataset names
     :param phedexUrl: a string with the PhEDEx URL
+    :return: a dictionary of datasets and their respective sizes
+    """
+    sizeByDset = {}
+    if not datasets:
+        return sizeByDset
+
+    urls = ['%s/blockreplicas?dataset=%s' % (phedexUrl, dset) for dset in datasets]
+    data = multi_getdata(urls, ckey(), cert())
+
+    for row in data:
+        dataset = row['url'].split('=')[-1]
+        rows = json.loads(row['data'])
```

Comment:
here you should account that row may not contain 'data' or be dict type and add a check before asking `json.loads` to parse the stream. The parsing may fail too, so wrap it up with appropriate try/except block.

Reply:
Ok, I see now that `data` can actually be `None` if the request fails:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/pycurl_manager.py#L469
I'll make sure `data` evaluates to True before doing anything.
About the return data type, I understand it will only change if we change the request headers. Provided we keep making the same requests to the same API, I think it's pretty safe not to check it (otherwise the code becomes quite unreadeable as well with so many try/except blocks)

Reply:
Alan, since you pointed out to pycurl_manager code, the returned message contains url, data and error and code. So you probably need to check for error/code and treat response correctly.